exporter.go

```diff
@@ -692,6 +702,44 @@ func parseConnectedSlaveString(slaveName string, slaveInfo string) (offset float
 	return
 }
 
+/*
+	valid examples:
+		master0:name=user03,status=sdown,address=192.169.2.52:6381,slaves=1,sentinels=5
+		master1:name=user02,status=ok,address=192.169.2.54:6380,slaves=1,sentinels=5
+*/
+func parseSentinelMasterString(master string, masterInfo string) (masterName string, masterStatus string, masterAddr string, masterSlaves float64, masterSentinels float64, ok bool) {
+	ok = false
+	if matched, _ := regexp.MatchString(`^master\d+`, master); !matched {
+		return
+	}
+	matchedMasterInfo := make(map[string]string)
+	for _, kvPart := range strings.Split(masterInfo, ",") {
+		x := strings.Split(kvPart, "=")
+		if len(x) != 2 {
+			log.Errorf("Invalid format for sentinel's master string, got: %s", kvPart)
+			continue
+		}
+		matchedMasterInfo[x[0]] = x[1]
+	}
+
+	masterName = matchedMasterInfo["name"]
+	masterStatus = matchedMasterInfo["status"]
+	masterAddr = matchedMasterInfo["address"]
+	masterSlaves, err := strconv.ParseFloat(matchedMasterInfo["slaves"], 64)
+	if err != nil {
+		log.Debugf("parseSentinelMasterString(): couldn't parse slaves value, got: %s, err: %s", matchedMasterInfo["slaves"], err)
+		return
+	}
+	masterSentinels, err = strconv.ParseFloat(matchedMasterInfo["sentinels"], 64)
+	if err != nil {
+		log.Debugf("parseSentinelMasterString(): couldn't parse sentinels value, got: %s, err: %s", matchedMasterInfo["sentinels"], err)
+		return
+	}
+	ok = true
```

Comment:
We can assign `ok` to `true` earlier, but I'm not sure is it really ok :)

Reply:
I think this is fine.